tests/unit/test_utils.py

```diff
@@ -0,0 +1,312 @@
+from pathlib import Path, PosixPath
```

Comment:
My personal preference would be to place this into `tests/` not `tests/unit` I think usually `tests` are considered units. WDYT? @sfc-gh-astus ?

Reply:
In java projects we always separated tests into `test/unit` and `test/integration`. 

In @sfc-gh-turbaszek approach we will have something like this:
- config (folder for unit)
- integration (folder with integrations)
- output (folder with units)
- some unit .py

It will be a mess for me

Reply:
I will move `tests/integration` to `tests-integration` and it will be fine :)